src/gtk/toplevel.cpp

```diff
@@ -1541,24 +1564,37 @@ void wxTopLevelWindowGTK::GTKUpdateDecorSize(const DecorSize& decorSize)
             }
             DoSetSizeHints(m_minWidth, m_minHeight, m_maxWidth, m_maxHeight, m_incWidth, m_incHeight);
         }
-        if (m_deferShow)
+
+        // We decide to defer showing the window only when its total (and not
+        // client) size had been set, and in this case we didn't set it
+        // correctly because we didn't have the correct decorations sizes when
+        // we did it -- but now that we do, we can adjust the size to the
+        // desired value.
+        //
+        // Note that we can't test for m_deferShow itself here because it may
+        // have been already reset to false if a WM had generated a
+        // notification with wrong _NET_REQUEST_FRAME_EXTENTS values first (as
+        // mutter does, at least in GNOME 3.48, where it sends 0 values for the
+        // not yet mapped window).
+        if (m_deferShowAllowed)
         {
+            const bool isResizeable = gtk_window_get_resizable(GTK_WINDOW(m_widget));
+
             // keep overall size unchanged by shrinking m_widget
-            int w, h;
-            GTKDoGetSize(&w, &h);
+            const wxSize size = GTKDoGetSize(isResizeable);
+
             // but not if size would be less than minimum, it won't take effect
-            if (w >= m_minWidth - (decorSize.left + decorSize.right) &&
-                h >= m_minHeight - (decorSize.top + decorSize.bottom))
+            if (size.x >= m_minWidth - (decorSize.left + decorSize.right) &&
+                size.y >= m_minHeight - (decorSize.top + decorSize.bottom))
             {
-                gtk_window_resize(GTK_WINDOW(m_widget), w, h);
-                if (!gtk_window_get_resizable(GTK_WINDOW(m_widget)))
-                    gtk_widget_set_size_request(GTK_WIDGET(m_widget), w, h);
-                resized = true;
+                gtk_window_resize(GTK_WINDOW(m_widget), size.x, size.y);
+                if (!isResizeable)
+                    gtk_widget_set_size_request(GTK_WIDGET(m_widget), size.x, size.y);
             }
         }
-        if (!resized)
```

Comment:
I'm pretty sure this still needed.

Reply:
OK, thinking more about this, I think you're right, I'll revert this change and will add a comment explaining why.